library/core/src/ffi/c_str.rs

```diff
@@ -547,7 +551,8 @@ impl CStr {
     /// assert_eq!(cstr.to_str(), Ok("foo"));
     /// ```
     #[stable(feature = "cstr_to_str", since = "1.4.0")]
-    pub fn to_str(&self) -> Result<&str, str::Utf8Error> {
+    #[rustc_const_unstable(feature = "const_cstr_to_bytes", issue = "none")]
+    pub const fn to_str(&self) -> Result<&str, str::Utf8Error> {
         // N.B., when `CStr` is changed to perform the length check in `.to_bytes()`
```

Comment:
Does this conflict with `to_*` methods being made `const`?

Reply:
I don't see how it would conflict

Reply:
The length check is currently done with `strlen` in `from_ptr` which can't be const I guess. If this would be moved to `to_bytes`, it would also not be const.

But I guess it would still be possible to make it const by not using `strlen` or by using `const_eval_select`.

Reply:
Yeah, right, we can do the same as we did with `memchr` in this PR